### PR TITLE
[minigraph.py]: Prefer parsing device type from <ElementType> 

### PR DESCRIPTION
**Pull request by theasianpianist**
Signed-off-by: Lawrence Lee <lawlee@microsoft.com>

**- Why I did it**
In some minigraph deployments, the `type` attribute in the `<Device>` tag (`<PngDec> -> <Devices>`) sometimes includes the XML namespace (e.g. instead of "BackEndLeafRouter" the value is "a:BackEndLeafRouter").

**- How I did it**
When parsing devices, prefer the `<ElementType>` tag usually found inside of `<Device>`, and use the `type` attribute as a fallback in case it is not found.

**- How to verify it**
Run the sonic-config-engine tests

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**
image

### CODE REVIEW

**Comment by prsunny** [comment]
@theasianpianist , can you please update the PR heading, it doesn't match the changes?

**Comment by theasianpianist** [comment]
retest baseimage please

**Review by prsunny — APPROVED** [review]
lgtm, please wait for @lguohan's approval

**Comment by theasianpianist** [comment]
retest vsimage please

**Comment by lguohan** [comment]
I remember the name space is different

**Comment by theasianpianist** [comment]
> I remember the name space is different

I'm not sure I understand, do you mean the namespace used in `minigraph.py` to parse the type? Or the namespace used in the sample minigraph?

**Comment by theasianpianist** [comment]
retest vsimage please

**Comment by lguohan** [comment]
```
Device xmlns:a="http://schemas.datacontract.org/2004/07/Microsoft.Search.Autopilot.Evolution" i:type="a:BackEndLeafRouter">
<ElementType>BackEndLeafRouter</ElementType>
<Address xmlns:b="Microsoft.Search.Autopilot.NetMux">
</Address>
```

**Comment by lguohan** [comment]
here is what i see in the example, it seems different from you test example.

**Comment by lguohan** [comment]
it looks like you are using a different approach to address the proble, which is fine. wonder if you can update the sample graph to inlcude the proper namespace so that we can make sure it address the issue.

**Comment by theasianpianist** [comment]
I will update it for consistency. However I don't believe the specific namespace included in the `type` attribute is important, since the entire attribute is treated as a string by the XML parser.

**Review by lguohan — APPROVED** [review]

**Review by prsunny — APPROVED** [review]

**Pull request merged**
